sphinx/texinputs/sphinxhowto.cls

```diff
@@ -92,7 +92,7 @@
 % so no page break before it.
 \newenvironment{sphinxthebibliography}[1]{%
   \begin{thebibliography}{1}%
-  \addcontentsline{toc}{section}{\bibname}}{\end{thebibliography}}
+  \addcontentsline{toc}{section}{\ifx\bibname\bibname\else\refname\fi}}{\end{thebibliography}}
```

Comment:
constraints:

- babel defines `\bibname` even if not used by document class
- polyglossia does not
- Japanese document are by default with `jreport.cls` and want `\bibname`
- non-Japanese document use `article.cls` and want `\refname`
- so far Sphinx used always `\bibname` but this was a bug.

I propose

    \ifdefined\refname\refname\else\ifdefined\bibname\bibname\fi\fi

At worst this conditional creates empty name, but it can not create compilation error (like we had with `\bibname` undefined due to using `polyglossia` and not `babel` anymore).